### PR TITLE
[Feature][array-type] Support ArrayLiteral in SQL.

### LINKED ISSUE #8074: [Feature] [array-type] Support ArrayLiteral in SQL.
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

This issue is a sub task of #7570 .

Support ArrayLiteral in SQL. 
An array literal is a list of zero or more expressions, each of which represents an array element, enclosed in square brackets (eg. [1, 2, 3]).

### Use case

```shell
mysql> create table array_type_table(k1 INT, k2 Array<int>) duplicate key (k1)
    -> distributed by hash(k1) buckets 1 properties('replication_num' = '1');
mysql> insert into array_type_table values (1, [1, 2, 3]);
```

### Related issues

#7570 

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by adonis0147**
# Proposed changes

Issue Number: close #8074 .

## Problem Summary:

Please refer to #8074 .

## Checklist(Required)

1. Does it affect the original behavior: No.
2. Has unit tests been added: Yes.
3. Has document been added or modified: No.
4. Does it need to update dependencies: No.
5. Are there any changes that cannot be rolled back: No.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/ArrayType.java [hunk 1]
```diff
@@ -17,16 +17,15 @@
 
 package org.apache.doris.catalog;
 
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.gson.annotations.SerializedName;
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/ArrayType.java:22]: the import order is
```
org.apache.doris
<blank line>
com
<blank line>
org
<blank line>
java
<blank line>
javax
```

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java [hunk 2]
```diff
@@ -159,6 +159,7 @@ public Column(Column column) {
     public void createChildrenColumn(Type type, Column column) {
         if (type.isArrayType()) {
             Column c = new Column(COLUMN_ARRAY_CHILDREN, ((ArrayType) type).getItemType());
+            c.setIsAllowNull(true);
```
Comment (yangzhg) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java:162]: why always allow null

File: fe/fe-core/src/main/java/org/apache/doris/catalog/ArrayType.java (hunk 1 above)
Comment (adonis0147) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/ArrayType.java]: Ok, I will fix it.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java (hunk 2 above)
Comment (adonis0147) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java:164]: I'm not sure whether there is a way to declare the item type in array type nullable.

Comment (yangzhg) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java:164]: add a todo comment to his

Comment (adonis0147) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/Column.java:167]: Done.

**Review by yangzhg — APPROVED** [review]
LGTM

**Pull request merged**